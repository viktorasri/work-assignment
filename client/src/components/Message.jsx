import React from 'react';
import PropTypes from 'prop-types';

const Message = ({ children }) => {
  return <div className='message message--red'>{children}</div>;
};

Message.propTypes = {
  children: PropTypes.oneOfType([PropTypes.element, PropTypes.string]).isRequired,
};

export default Message;
