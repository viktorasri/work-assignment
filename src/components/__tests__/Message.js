import React from 'react';
import { shallow } from 'enzyme';
import Message from '../Message';

describe('<Message />', () => {
  const wrapper = shallow(<Message>hello</Message>);

  it('expect to render component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('expect to pass a variant of "blue" and render string', () => {
    const wrapperBlue = shallow(<Message variant='blue'>hello</Message>);
    expect(wrapperBlue.find('.message--blue')).toHaveLength(1);
    expect(wrapperBlue.text()).toEqual('hello');
  });
});
