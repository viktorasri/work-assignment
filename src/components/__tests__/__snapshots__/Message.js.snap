// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Message /> expect to render component 1`] = `
<div
  className="message message--red"
>
  hello
</div>
`;
